### PR TITLE
Amazon support

### PR DESCRIPTION
**Pull request by vegaro**

### CODE REVIEW

**Comment by NachoSoto** [comment]
What's the status of this?

File: src/purchases.ts [hunk 1]
```diff
@@ -187,13 +196,14 @@ export default class Purchases {
...
      * Default is null, which will make the SDK use standardUserDefaults.
      */
-    public static setup(
-        apiKey: string,
-        appUserID?: string | null,
-        observerMode: boolean = false,
-        userDefaultsSuiteName?: string,
-        usesStoreKit2IfAvailable: boolean = false
-    ): void {
+    public static setup({
```
Comment (NachoSoto) [inline src/purchases.ts:199]: I've renamed this to `configure` in #374.

Comment (vegaro) [reply src/purchases.ts:199]: Note this version uses a `PurchasesConfiguration` object as a single parameter, I am not a JS expert but I think this is a common practice when the functions have a lot of optional arguments

**Comment by NachoSoto** [comment]
Rebased and reopened in #393.

**Pull request closed without merging**